### ISSUE TITLE
InputText with a property that has a MaxLength data annotation should generate a maxlength attribute

### ISSUE THREAD

**Issue by csharpfritz** [dotnet/aspnetcore#48246]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

I am using the `InputText` element in Blazor and would like it to limit the number of characters allowed to be keyed in based off of the `MaxLength` data annotation 

### Describe the solution you'd like

Generate a maxlength attribute on the resulting HTML input element that contains the value of the MaxLength data annotation

**Comment by mkArtakMSFT** [comment]
Related: https://github.com/dotnet/aspnetcore/issues/16650

**Comment by csharpfritz** [comment]
Is this something I could submit a PR for?JeffOn Nov 26, 2023, at 13:51, msftbot[bot] ***@***.***> wrote:﻿Re: [dotnet/aspnetcore] InputText with a property that has a MaxLength data annotation should generate a maxlength attribute (Issue #48246)We've moved this issue to the Backlog milestone. This means that it is not going to be worked on for the coming release. We will reassess the backlog following the current release and consider this item at that time. To learn more about our issue management process and to have better expectation regarding different types of issues you can read our Triage Process.—Reply to this email directly, view it on GitHub or unsubscribe.You are receiving this email because you authored the thread.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by javiercn** [comment]
I don't believe this is something we want to do automatically out of the box.

We don't translate data annotations to the underlying HTML attributes (and I believe MVC doesn't either).

I believe this changes the validation experience, maxlength I believe simply precludes the user from typing additional characters as opposed to triggering validation afterwards.

That can be a jarring experience, especially when copying and pasting from the clipboard.

Should we ever decide to do something like this, I believe we would analyze all the relevant attributes and do something explicit for all of them rather than a one off.

I'm going to close the issue for now since we don't plan to take action in this direction.
